executor/error.rs

```diff
@@ -17,5 +18,8 @@ typedb_error!(
         CreatingIterator(3, "Error creating iterator from {instruction_name} instruction.", instruction_name: String, ( source: ConceptReadError )),
         AdvancingIteratorTo(4, "Error moving iterator (by steps or seek) to target value.", ( source: ConceptReadError )),
         ExpressionEvaluate(5, "Error evaluating expression", ( source: ExpressionEvaluationError )),
+        TabledFunctionLockError(6, "The table function lock was poisoned", function_id: FunctionID, arguments: MaybeOwnedRow<'static>),
```

Comment:
When can that happen that isn't an immediate `abort`?

Reply:
Can't. May as well unwrap it then?